src/include/platform/ConfigurationManager.h

```diff
@@ -99,9 +101,14 @@ class ConfigurationManager
     virtual CHIP_ERROR GetSpake2pIterationCount(uint32_t & iterationCount)                          = 0;
     virtual CHIP_ERROR GetSpake2pSalt(uint8_t * buf, size_t bufSize, size_t & saltLen)              = 0;
     virtual CHIP_ERROR GetSpake2pVerifier(uint8_t * buf, size_t bufSize, size_t & verifierLen)      = 0;
-    // Lifetime counter is monotonic counter that is incremented only in the case of a factory reset
-    virtual CHIP_ERROR GetLifetimeCounter(uint16_t & lifetimeCounter)                  = 0;
-    virtual CHIP_ERROR IncrementLifetimeCounter()                                      = 0;
+#if CHIP_ENABLE_ROTATING_DEVICE_ID
+    // Lifetime counter is monotonic counter that is incremented upon each commencement of advertising
+    virtual CHIP_ERROR GetLifetimeCounter(uint16_t & lifetimeCounter) = 0;
+    virtual CHIP_ERROR IncrementLifetimeCounter()                     = 0;
+    // Unique ID is identifier utilized for the rotating device ID calculation purpose as an input key. It is separate identifier
+    // from the Basic cluster unique ID.
+    virtual CHIP_ERROR GetRotatingDeviceIdUniqueId(uint8_t * buf, size_t & outSize, size_t bufSize) = 0;
```

Comment:
All the other getters in this file have bufSize first, then the output length.....

_Ideally_ this would just take a `MutableByteSpan` by non-const reference, so we can't lose track of what sizes go with what...

Reply:
Done